### PR TITLE
Zend_Db_Select: add SQL NOWAIT and SKIP LOCKED options for SELECT ... FOR UPDATE

### PR DESCRIPTION
**Pull request by thomaslauria**
Zend_Db_Select was missing the ability to define a FOR UPDATE statement with SQL NOWAIT or SKIP LOCKED options.
These are added now. The forUpdate method still accepts boolean, but additionally the flags to configure the options.

### CODE REVIEW

**Comment by thomaslauria** [comment]
Any chance to integrate that in upstream?

**Comment by Shardj** [comment]
Just waiting on the the pending review then I'll merge this

**Comment by Shardj** [comment]
I'll probably close this soon if it's just going to sit here otherwise

**Comment by thomaslauria** [comment]
Ok, close means merge or reject? In the latter case: what is missing?

**Comment by Shardj** [comment]
They're very minor, shouldn't take more than a minute

**Comment by glensc** [comment]
I don't see any review notes. how to see them?

**Comment by Shardj** [comment]
Sorry, I didn't realise that the review wasn't published

File: library/Zend/Db/Select.php [hunk 1]
```diff
@@ -1318,6 +1329,12 @@ protected function _renderForupdate($sql)
             $sql .= ' ' . self::SQL_FOR_UPDATE;
         }
 
+        if ($this->_parts[self::FOR_UPDATE_MODE] === self::FU_MODE_NOWAIT) {
+            $sql .= ' ' . self::SQL_FU_NOWAIT;
+        }elseif($this->_parts[self::FOR_UPDATE_MODE] === self::FU_MODE_SKIP) {
```
[the remark this answers is not in the document]
Comment (thomaslauria) [reply library/Zend/Db/Select.php:1334]: for sure...

File: library/Zend/Db/Select.php [hunk 2]
```diff
@@ -667,14 +675,17 @@ public function limitPage($page, $rowCount)
...
      *
-     * @param bool $flag Whether or not the SELECT is FOR UPDATE (default true).
+     * @param mixed $flag Whether or not the SELECT is FOR UPDATE (default true), pass the flag FU_MODE_NOWAIT or FU_MODE_SKIP to make the FOR UPDATE either NOWAIT or SKIP LOCKED
      * @return Zend_Db_Select This Zend_Db_Select object.
      */
     public function forUpdate($flag = true)
     {
-        $this->_parts[self::FOR_UPDATE] = (bool) $flag;
+        $this->_parts[self::FOR_UPDATE] = (bool) $flag; //still use bool here for backwards compatibility
+        if($flag === self::FU_MODE_NOWAIT || $flag === self::FU_MODE_SKIP) {
```
[the remark this answers is not in the document]
Comment (thomaslauria) [reply library/Zend/Db/Select.php]: done

**Comment by thomaslauria** [comment]
> Sorry, I didn't realise that the review wasn't published

No problem ;)

**Pull request merged**